migrations/archive/2020/20201111_api_date.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Fix dates in the database that were stored as $type string instead of Date
+ */
+/* eslint-disable no-console */
+
+const MIGRATION_NAME = '20201111_api_date';
+
+import * as Tasks from '../website/server/models/task';
+
+const progressCount = 1000;
+let count = 0;
+
+async function updateUser (user) {
+  count++;
+
+  let set = {};
+  const newDate = new Date(user.date);
+  if (isValidDate(newDate)) {
+    set = {
+      date: newDate
+    }
+  } else {
+    set = {
+      date: undefined
+    }
+  };
+
+  if (count % progressCount === 0) console.warn(`${count} ${user._id}`);
+  return await Tasks.Task.update({_id: user._id}, {$set: set}).exec();
+}
+
+module.exports = async function processUsers () {
+  let query = {
+    date: {$exists: true}
```

Comment:
@thewilloftheshadow can we limit the query to `type: 'todo'`? the date field exists only for them